src/settings.rs

```diff
@@ -36,7 +36,265 @@ pub struct FacetingSettings {
+/// use serde_json::Value;
+/// let embedder_setting = GenericRestEmbedderSettings {
+///   url: Some("http://localhost:12345/api/v1/embed".to_string()),
+///   api_key: Some("SOURCE_API_KEY".to_string()),
+///   dimensions: Some(512),
+///   document_template: Some("A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}".to_string()),
+///   request: HashMap::from([
+///     ("model".to_string(), Value::from("MODEL_NAME")),
+///     ("prompt".to_string(), Value::from("{{text}}"))
+///   ]),
+///   response: HashMap::from([
+///     ("model".to_string(), Value::from("{{embedding}}"))
+///   ]),
+///   headers: HashMap::from([
+///     ("X-MAGIC".to_string(), "open sesame".to_string())
+///   ]),
+/// };
+/// # let expected = serde_json::json!({
+/// #   "url":"http://localhost:12345/api/v1/embed",
+/// #   "apiKey":"SOURCE_API_KEY",
+/// #   "dimensions":512,
+/// #   "documentTemplate":"A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}",
+/// #   "request":{"prompt":"{{text}}","model":"MODEL_NAME"},
+/// #   "response":{"model":"{{embedding}}"},
+/// #   "headers":{"X-MAGIC":"open sesame"}
+/// # });
+/// # let expected: GenericRestEmbedderSettings = serde_json::from_value(expected).unwrap();
+/// # assert_eq!(embedder_setting, expected);
+/// ```
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericRestEmbedderSettings {
+    /// Mandatory, full URL to the embedding endpoint
+    /// Must be parseable as a URL.
+    /// If not specified, [Meilisearch](https://www.meilisearch.com/) (**not the sdk you are currently using**) will try to fetch the `MEILI_OLLAMA_URL` environment variable
+    /// Example: `"http://localhost:12345/api/v1/embed"`
+    pub url: Option<String>,
+    /// Optional, passed as Bearer in the Authorization header
+    /// Example: `"187HFLDH97CNHN"`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub api_key: Option<String>,
+    /// Optional
+    /// Inferred with a dummy request if missing
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dimensions: Option<usize>,
+    /// Use it to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+    ///
+    /// if present, `document_template` must be a [Liquid template](https://shopify.github.io/liquid/).
+    /// Use `{{ doc.attribute }}` to access document field values.
+    /// Meilisearch also exposes a `{{ fields }}` array containing one object per document field, which you may access with `{{ field.name }}` and `{{ field.value }}`.
+    ///
+    /// For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.
+    /// Example: `"A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"`
```

Comment:
```suggestion
    /// Example: `"A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"`
    /// 
    /// Default:
    /// ```raw
    /// {% for field in fields %}
    /// {% if field.is_searchable and not field.value == nil %}
    /// {{ field.name }}: {{ field.value }}\n
    /// {% endif %}
    /// {% endfor %}
    /// ```
```